### PR TITLE
[Shell] expose CurrentShell

### LINKED ISSUE #4625: [Shell] statically expose the current shell
## Summary
Getting a reference to the current shell is tedious and repetitive. As the shell has to be the App.MainPage, we could expose a static version of `GoToAsync()` that tries to get the shell from there. If the shell is not found, an exception will be thrown.

If the user want to navigate on a shell that's not currently App.MainPage (multiple shell app, anyone ?) he can still keep a reference to that shell and use the non-static GoToAsync

this is required for #4594

## API Changes

```csharp
public Task static Shell.GoToAsync(ShellNavigationState state, bool animate = true);
```
## Intended Use Case

replace this usage
```csharp
var shell = Application.Current.MainPage as Shell;
await shell.GoToAsync("app:///s/about?welcome=you");
```

by
```csharp
await Shell.GoToAsync("app:///s/about?welcome=you");
```

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

[Shell] expose CurrentShell

### Issues Resolved ### 

- fixes #4625

### API Changes ###

 
 None

### Platforms Affected ### 

- Core/XAML

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Core/Shell/Shell.cs [hunk 1]
```diff
@@ -324,9 +322,7 @@ void IShellController.UpdateCurrentState(ShellNavigationSource source)
 			OnNavigated(new ShellNavigatedEventArgs(oldState, CurrentState, source));
 		}
 
-#endregion IShellController
-
-#region URI Navigation
+		public static Shell CurrentShell => Application.Current?.MainPage as Shell ?? throw new InvalidOperationException("Application.Current.MainPage is not a Shell");
```
Comment (paymicro) [inline Xamarin.Forms.Core/Shell/Shell.cs]: is it necessary to throw an exception here?

File: Xamarin.Forms.Core/Shell/Shell.cs [hunk 2]
```diff
@@ -162,8 +162,6 @@ static void OnShellColorValueChanged(BindableObject bindable, object oldValue, o
 
 		static readonly BindablePropertyKey MenuItemsPropertyKey =
 			BindableProperty.CreateReadOnly(nameof(MenuItems), typeof(MenuItemCollection), typeof(Shell), null, defaultValueCreator: bo => new MenuItemCollection());
-			
-#region IShellController
```
Comment (hartez) [inline Xamarin.Forms.Core/Shell/Shell.cs]: Why drop the regions?

File: Xamarin.Forms.Core/Shell/Shell.cs (hunk 1 above)
Comment (hartez) [reply Xamarin.Forms.Core/Shell/Shell.cs]: I'm wondering the same thing. Do we have any mechanism for a component to determine if it's running in a Shell app (as opposed to a classic app)? If not, does it make sense to allow a check for `CurrentShell == null`?

File: Xamarin.Forms.Core/Shell/Shell.cs (hunk 2 above)
Comment (StephaneDelcroix) [reply Xamarin.Forms.Core/Shell/Shell.cs]: for a few reasons:
- I don't like them
- they're not consistent with the rest of the code, or even the rest of the shell code
- they enforce some weird code ordering (this one had a region for BP, then a region for interface implementation. those 2, for example, can collide)
- they're out of date faster than code comments

now, I get why one might like them: being able to collapse region. I know a thing or two about where to place opening braces, that saves SO MUCH vertical space that collapsing is no longer required AT ALL.

File: Xamarin.Forms.Core/Shell/Shell.cs (hunk 1 above)
Comment (StephaneDelcroix) [reply Xamarin.Forms.Core/Shell/Shell.cs]: go one
```suggestion
		public static Shell CurrentShell => Application.Current?.MainPage as Shell;
```

File: Xamarin.Forms.Core/Shell/Shell.cs (hunk 2 above)
Comment (hartez) [reply Xamarin.Forms.Core/Shell/Shell.cs]: Fair enough. I didn't realize they weren't being applied consistently. 

> I know a thing or two about where to place opening braces, that saves SO MUCH vertical space that collapsing is no longer required AT ALL.
 
:)

**Pull request merged**